api.go

```diff
@@ -440,6 +440,11 @@ func StartContainer(podID, containerID string) (VCContainer, error) {
 		return nil, err
 	}
 
+	// Before starting the container, POD's resources MUST BE updated
+	if err := p.updateResources(); err != nil {
```

Comment:
Ok let me try to explain here. This call should be called from the function `createContainer()` in `container.go`. This way, it will be called anytime a container needs to be created. Indeed, this function is called after a pod has been created (VM has been started) in order to create new containers if they are provided through the PodConfig, or it is called from `CreateContainer()` here in `api.go`, where we know the VM is already running.
More than factorization, I think it is important to perform the hotplug as soon as possible since the container is really created at the create time, meaning a process is really running inside the agent in case of Kata Containers. And this tend to be the real way to handle this.